modules/polyfills/src/fetch-node/utils/stream-utils.node.js

```diff
@@ -8,9 +9,23 @@ export function concatenateReadStream(readStream) {
       if (typeof chunk === 'string') {
         string += chunk;
       } else {
-        arrayBuffer = concatenateArrayBuffers(arrayBuffer, chunk);
+        switch (readStream.headers['content-encoding']) {
+          case 'br':
+            arrayBuffer = concatenateArrayBuffers(arrayBuffer, zlib.brotliDecompressSync(chunk));
+            break;
+          case 'gzip':
+            arrayBuffer = concatenateArrayBuffers(arrayBuffer, zlib.gunzipSync(chunk));
```

Comment:
I am a little uncertain on whether this works with multiple chunks without any additional flags/calls. 

This would be an issue on larger files, if at all. Worth taking a second look at the docs at https://nodejs.org/dist/latest-v14.x/docs/api/zlib.html#zlib_compressing_http_requests_and_responses.

The main concern being that one raw memory chunk that is cut at an arbitrary byte might not be "fully consumed" and some residual state is needed when parsing the next chunk.